tools/hess/hess_astro_tool.xml

```diff
@@ -0,0 +1,138 @@
+  <inputs>
+    <conditional name="_data_product">
+      <param name="_selector" type="select" label="Data Product">
+        <option value="Image" selected="true">Image</option>
+        <option value="Spectrum" selected="false">Spectrum</option>
+        <option value="Lightcurve" selected="false">Lightcurve</option>
+      </param>
+      <when value="Image">
+        <param name="src_name" type="text" value="Crab" label="src_name" />
+        <param name="RA" type="float" value="83.6287" label="RA Units: deg" />
+        <param name="DEC" type="float" value="22.0147" label="DEC Units: deg" />
+        <param name="T1" type="text" value="2000-10-09T13:16:00.0" label="T1" />
+        <param name="T2" type="text" value="2022-10-10T13:16:00.0" label="T2" />
+        <param name="Radius" type="float" value="2.5" label="Radius Units: deg" />
+        <param name="pixsize" type="float" value="0.1" label="pixsize Units: deg" />
+        <param name="Emin" type="float" value="100.0" label="Emin Units: GeV" />
+        <param name="Emax" type="float" value="10000.0" label="Emax Units: GeV" />
+      </when>
+      <when value="Spectrum">
+        <param name="src_name" type="text" value="Crab" label="src_name" />
+        <param name="RA" type="float" value="83.6287" label="RA Units: deg" />
+        <param name="DEC" type="float" value="22.0147" label="DEC Units: deg" />
+        <param name="T1" type="text" value="2000-10-09T13:16:00.0" label="T1" />
+        <param name="T2" type="text" value="2022-10-10T13:16:00.0" label="T2" />
+        <param name="Radius" type="float" value="2.5" label="Radius Units: deg" />
+        <param name="R_s" type="float" value="0.2" label="R_s Units: deg" />
+        <param name="Emin" type="float" value="100.0" label="Emin Units: GeV" />
+        <param name="Emax" type="float" value="10000.0" label="Emax Units: GeV" />
+        <param name="NEbins" type="integer" value="20" label="NEbins" />
+      </when>
+      <when value="Lightcurve">
+        <param name="src_name" type="text" value="Crab" label="src_name" />
+        <param name="RA" type="float" value="83.6287" label="RA Units: deg" />
+        <param name="DEC" type="float" value="22.0147" label="DEC Units: deg" />
+        <param name="T1" type="text" value="2003-10-09T13:16:00.0" label="T1" />
+        <param name="T2" type="text" value="2005-10-10T13:16:00.0" label="T2" />
+        <param name="Radius" type="float" value="2.5" label="Radius Units: deg" />
+        <param name="R_s" type="float" value="0.2" label="R_s Units: deg" />
+        <param name="Emin" type="float" value="100.0" label="Emin Units: GeV" />
+        <param name="Emax" type="float" value="10000.0" label="Emax Units: GeV" />
+        <param name="NTbins" type="integer" value="2" label="NTbins" />
+      </when>
+    </conditional>
+  </inputs>
+  <outputs>
+    <data label="${tool.name} -&gt; Image picture" name="out_Image_picture" format="auto" from_work_dir="picture_galaxy.output">
```

Comment:
I know format="auto" is the most simple solution and is good in the beginning. I just wanted to point out that is would be nice and also faster if we could specify a fixed value here.

Reply:
That's the best general way I was able to came up with. As far as tool description is generated from the notebook, and we can't be 100% sure of the outputs format without actually running it